src/SharpCompress/Archives/SevenZip/SevenZipArchive.cs

```diff
@@ -101,11 +101,23 @@ protected override IEnumerable<SevenZipArchiveEntry> LoadEntries(IEnumerable<Sev
         {
             var stream = volumes.Single().Stream;
             LoadFactory(stream);
+            var entries = new SevenZipArchiveEntry[database._files.Count];
             for (int i = 0; i < database._files.Count; i++)
             {
                 var file = database._files[i];
-                yield return new SevenZipArchiveEntry(this, new SevenZipFilePart(stream, database, i, file, ReaderOptions.ArchiveEncoding));
+                entries[i] = new SevenZipArchiveEntry(this, new SevenZipFilePart(stream, database, i, file, ReaderOptions.ArchiveEncoding));
```

Comment:
I like my yield returns for streaming but 7Zip always requires random access so no big deal

Reply:
I just realised

`                foreach (var entry in group)
                {
                    entry.IsSolid = isSolid;
                    isSolid = true; //mark others in this group as solid - same as rar behaviour.
                    yield return entry;
                }
`

Reply:
maybe being a big more clever is warranted but 7Zip is my most hated format so I'm ambivalent.